advanced-solidus/images-and-image-processing.md

```diff
@@ -0,0 +1,196 @@
+  extend ActiveSupport::Concern
+  include ::Spree::ActiveStorageAdapter
+
+  delegate :width, :height, to: :attachment, prefix: true
+
+  included do
+    validates :attachment, presence: true
+    validate :attachment_is_an_image
+    validate :supported_content_type
+
+    has_attachment :attachment,
+                   styles: {
+                    mini: '48x48>',
+                    small: '400x400>',
+                    product: '680x680>',
+                    large: '1200x1200>',
+                    jumbo: '1600x1600>'
+                   },
+                   default_style: :product
+
+    def supported_content_type
+      unless attachment.content_type.in?(::Spree::Config.allowed_image_mime_types)
+        errors.add(:attachment, :content_type_not_supported)
+      end
+    end
+  end
+end
+```
+{% endtab %}
+
+However, to make the module override work, we will also have to instruct Spree to utilize our customized `my_active_storage_attachment` instead of the default:
+
+{% code title="config/initializers/spree.rb" %}
+```ruby
+Spree.config do |config|
+  # ...
+  config.image_attachment_module = 'AmazingStore::Spree::Image::ActiveStorageAttachment'
+  # ...
+end
+```
+{% endcode %}
+
+After appending these new styles into settings, they can be utilized in your store by passing the style when rendering the image partial. Ex:
+
+{% code title="" %}
+```ruby
+<%= render 'spree/admin/shared/image', image: product.gallery.images.first, size: :jumbo %>
+```
+{% endcode %}
+
+## Modifying Variant Options
+### ActiveStorage
+
+MiniMagick allows the processing of variants with many [options](https://github.com/janko/image_processing/blob/master/doc/minimagick.md#imageprocessingminimagick) to customize your images. To take full advantage of these options, we will need to inject our modified method for `variant`. This can be accomplished with an **override** to Spree's current `variant` method utilizing Rail's suggested pattern for [improving engine functionality](https://guides.rubyonrails.org/engines.html#improving-engine-functionality).
+
+Solidus version 3.1 and greater is currently is set up to implement overrides by searching for files located in the overrides' directory found in your application's `root/app` folder. For our new variant method to take precedence, we will need to also instruct our override to `prepend` the attachment class' `variant` method. For Solidus versions below 3.1, the decorators' folder (`root/app/decorators`) is utilized instead. Let's assume all of our pictures are somehow imported upside down and cropped in a different shape. To correct this, we want to rotate all of them to be right side up and framed to the appropriate size when displayed in the view. Additionally, it was decided by your team that you wanted to compress and force convert your images png to reduce load times. This could be implemented by creating our new override with the following:
```

Comment:
> Solidus version 3.1 and greater is currently is set up to implement overrides by searching for files located in the overrides' directory found in your application's `root/app` folder. For our new variant method to take precedence, we will need to also instruct our override to `prepend` the attachment class' `variant` method. For Solidus versions below 3.1, the decorators' folder (`root/app/decorators`) is utilized instead.

Again, we should have a page/section for this stuff, let's just link to it.

Reply:
Agreed:
Utilizing learning examples on **overrides** from [customizing the core guide](customization/customizing-the-core.md#L232) again, we can implement our required changes without duplicating the entire class.